snuba/clickhouse/optimize/optimize.py

```diff
@@ -287,41 +328,71 @@ def optimize_partition_runner(
     2. The final cutoff time is reached. In this case, the scheduler will
     raise an exception which would be propagated to the caller.
     """
+
     remaining_partitions = partitions
     while remaining_partitions:
         schedule = scheduler.get_next_schedule(remaining_partitions)
-        num_threads = len(schedule.partitions)
+        scheduled_partition_and_jitter_pairs = _get_partition_and_jitter_pairs(schedule)
```

Comment:
> Also, how does this variable relate to `remaining_partitions`?
My understanding is that if there are any partitions from the current schedule that didn't finish optimizing, then they will get optimized again because `remaining_partitions` won't be empty

Reply:
Do we need two while loops to get that done? Do we need to re-create the threadpool in one of those loops? I would have expected that we'd just create the threadpool once.